### DISCUSSION TITLE
Is mapping ctrl+c to `copy_and_clear_or_interrupt` supposed to disable "scroll to end"

### DISCUSSION THREAD

**Discussion by 0xAdk** [kovidgoyal/kitty#4787] [Q&A] [answered]
I often use `ctrl+c` to return to the bottom of the screen after I've scrolled up, but after mapping it to `copy_and_clear_or_interrupt` It no longer scrolled to the begging when pressed. 

I was under the assumption that  `copy_and_clear_or_interrupt` would copy if there was a selection and otherwise work like the default `ctrl+c`

I'm currently using a work around which is mapping `ctrl+c` to `combine : copy_and_clear_or_interrupt : scroll_end`

Is this the Intended behavior?

**Comment by page-down** [comment, marked as answer]
I checked the code and it looks like `child_monitor.needs_write` sends `\x03` without the side effect of a keypress event.
It doesn't scroll scrollback to the bottom.

keys.c
```c
if (screen->scrolled_by && action == GLFW_PRESS && !is_modifier_key(key)) {
    screen_history_scroll(screen, SCROLL_FULL, false);  // scroll back to bottom
}
```

PR:
https://github.com/kovidgoyal/kitty/pull/4790

**Reply by 0xAdk** [reply]
wow, thanks for the fast turn around!

though if this wasn't the intended behavior shouldn't you patch other functions that call `child_monitor.needs_write`?
such as `send_text`?

**Reply by page-down** [reply]
Not all of them should be changed, and backwards compatibility should not be compromised.
If you think there is any other feature that is also problematic, you can discuss it separately.
